### ISSUE TITLE
Extendability

### ISSUE THREAD

**Issue by suhmantha1** [punkave/apostrophe-preferences#3]
Make this module able to be extended in the same way apostrophe-snippets is. You should be able to organize preferences at the project-level

**Comment by boutell** [comment]
Sam, can you be more specific about what you have in mind here? You can
already add any schema of fields you want.

On Wed, Jul 1, 2015 at 11:58 AM, Samantha Combs <notifications@github.com>
wrote:

> Make this module able to be extended in the same way apostrophe-snippets
> is. You should be able to organize preferences at the project-level
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/punkave/apostrophe-preferences/issues/3>.
>



-- 


*THOMAS BOUTELL, *DEV & OPS
P'UNK AVENUE | (215) 755-1330  |  punkave.com

**Comment by kylestetz** [comment]
Whoops! Sam and I talked about this offline. You can't subclass apostrophe-preferences into separate collections of preferences, but only because it is hardcoded to find only one preference object in the database.

**Comment by boutell** [comment]
So the goal was to be able to have many separate subclasses of preferences?

Interesting. Is the goal of that simply to attach them to separate buttons?

On Thu, Jul 2, 2015 at 9:15 AM, Kyle Stetz <notifications@github.com> wrote:

> Whoops! Sam and I talked about this offline. You can't subclass
> apostrophe-preferences into separate collections of preferences, but only
> because it is hardcoded to find only one preference object in the database.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/punkave/apostrophe-preferences/issues/3#issuecomment-118028583>
> .
>



-- 


*THOMAS BOUTELL, *DEV & OPS
P'UNK AVENUE | (215) 755-1330  |  punkave.com

**Comment by kylestetz** [comment]
Yep, or perhaps they become different menu items under the same dropdown. Design TBD.

**Comment by kylestetz** [comment]
Of course another option would be to use `groupFields`.

**Comment by boutell** [comment]
Yeah, that would be a lot simpler...

Unless you're looking for some sort of optimization like not actually
loading all of them all the time, groupFields seems a pretty natural
approach.

On Thu, Jul 2, 2015 at 9:20 AM, Kyle Stetz <notifications@github.com> wrote:

> Of course another option would be to use groupFields.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/punkave/apostrophe-preferences/issues/3#issuecomment-118030369>
> .
>



-- 


*THOMAS BOUTELL, *DEV & OPS
P'UNK AVENUE | (215) 755-1330  |  punkave.com
